engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -2757,18 +2760,22 @@ private void orchestrateMigrateAway(final String vmUuid, final long srcHostId, f
                 s_logger.debug("Unable to migrate VM due to: " + e.getMessage());
             }
 
```

Comment:
I know this wasn't part of our change... but shouldn't the `while (true)` loop from line 2733 end here?  If it did, it could try all possible destination hosts before giving up and possibly stopping the VM.

Reply:
Also, what is the point of `excludes.addHost(dest.getHost().getId());` on line 2753 if upon the first migration error we just try to stop the VM?  Should we try all available destinations and then if that fails we resort to possibly stopping the VM?

Reply:
For line 2750 `throw new InsufficientServerCapacityException("Unable to find a server to migrate to.", host.getClusterId());` if we did make this loop through available destinations, we should change this to:
 `if (excludes.length = 1) {
    throw new InsufficientServerCapacityException("Unable to find a server to migrate to.", host.getClusterId());
} else {
    break;
}` 

that way if we loop through all destinations and still don't successfully migrate we fall through to the possible stop of the VM instead of throwing the insufficient server capacity exception.